src/app/legacy/containers/PageHandlers/withRUM/index.jsx

```diff
@@ -63,9 +64,12 @@ const RUMLoader = Component => {
         'arn:aws:iam::923061562593:role/RUM-Monitor-eu-west-1-923061562593-2635993079561-Unauth',
     };
 
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const isOperaMini = useOperaMiniDetection();
```

Comment:
I don't understand why this lint rule triggered here, I feel this is a functional react component, would love some help if I've done this wrong?

Reply:
I think this is because hook usage needs to be declared at the highest level of the function, they can't exist within an if statement block, because of the React rendering strategy, hook usage needs to be pure

Reply:
There's a nested function here called withRum, which is not a React function component. If you move your check to outside of withRum, eslint won't complain

Reply:
Moving the hook outside of `withRum` caused a runtime error as the `RumLoader` is just a function returning anothe react component, not a funcitonal react component itself

https://github.com/bbc/simorgh/pull/10373/files#diff-ddd236da434bde238fa4a12aa7c954e90ae577538c305aa4e9f15de9694257d4L46

I've introduced a new react component to resolve that problem